src/mon/MonMap.cc

```diff
@@ -196,7 +196,11 @@ void MonMap::encode(ceph::buffer::list& blist, uint64_t con_features) const
   if (!HAVE_FEATURE(con_features, MONENC) ||
       !HAVE_FEATURE(con_features, SERVER_NAUTILUS)) {
     for (auto& [name, info] : mon_info) {
-      legacy_mon_addr[name] = info.public_addrs.legacy_addr();
+      auto addr = info.public_addrs.legacy_addr();
```

Comment:
Add a comment saying
```
// see note in mon_info_t::encode()
```
or something like that above this line.

Reply:
comment added.